Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,378 @@
+            }
+
+            foreach (var item in list)
+            {
+                if (item.Status.IsOpen())
+                {
+                    var cached = CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Any())
+                    {
+                        CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+
+            return list;
+        }
+
+        /// <summary>
+        /// Gets all open positions
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            var endpoint = GetEndpoint("positions");
+            var request = new RestRequest(endpoint, Method.POST);
+
+            JsonObject payload = new JsonObject();
+            payload.Add("request", endpoint);
+            payload.Add("nonce", GetNonce().ToString());
+
+            request.AddJsonBody(payload.ToString());
+            SignRequest(request, payload.ToString());
+
+            var response = ExecuteRestRequest(request);
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                throw new Exception($"BitfinexBrokerage.GetAccountHoldings: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+            }
+
+            var positions = JsonConvert.DeserializeObject<Messages.Position[]>(response.Content);
+            return positions.Where(p => p.Amount != 0)
+                .Select(ConvertHolding)
+                .ToList();
+        }
+
+        /// <summary>
+        /// Gets the total account cash balance for specified account type
+        /// </summary>
+        /// <returns></returns>
+        public override List<Cash> GetCashBalance()
+        {
+            var list = new List<Cash>();
+            var endpoint = GetEndpoint("balances"); ;
+            var request = new RestRequest(endpoint, Method.POST);
+
+            JsonObject payload = new JsonObject();
+            payload.Add("request", endpoint);
+            payload.Add("nonce", GetNonce().ToString());
+
+            request.AddJsonBody(payload.ToString());
+            SignRequest(request, payload.ToString());
+
+            var response = ExecuteRestRequest(request);
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                throw new Exception($"BitfinexBrokerage.GetCashBalance: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+            }
+
+            var availableWallets = JsonConvert.DeserializeObject<Messages.Wallet[]>(response.Content)
+                .Where(WalletFilter(_algorithm.BrokerageModel.AccountType));
+            foreach (var item in availableWallets)
+            {
+                if (item.Available > 0)
```

Comment:
I believe we should use `item.Amount` here (and in the two usages below) instead of `item.Available`, to include amounts reserved for open orders.